### PR TITLE
add debufr to sample utilities, along with corresponding test cases

### PR DESCRIPTION
**Pull request by jbathegit**
Part of #91 
Part of #32

### CODE REVIEW

File: utils/debufr.f [hunk 1]
```diff
@@ -0,0 +1,461 @@
+C> @file
```
Comment (aerorahul) [inline utils/debufr.f:1]: why C and fortran both?

Comment (jbathegit) [reply utils/debufr.f:1]: This utility has a lot of UNIX command line options, which at least historically are generally easier and more standard to process in C than in Fortran.  So the driver is written in C, and then most of the remaining code logic is in Fortran.  The library itself has a similar mix of C and Fortran, and as long as the interface across the languages is clean and deliberate, it works and is portable across all platforms.

**Comment by aerorahul** [comment]
I am going to let @edwardhartnett comment on this.

**Comment by jbathegit** [comment]
@edwardhartnett could you please take a look and let me know if you have any questions/concerns about this PR?  If not, I'd like to get it approved and be able to merge it, because I now have another testing-related issue (#104) to work on, and it will likely involve additional updates to some of the same configuration files.

Please let me know - thanks!

**Review by edwardhartnett — APPROVED** [review]

**Pull request merged**